Code/Source/Imu/ROS2ImuSensorComponent.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "Imu/ROS2ImuSensorComponent.h"
+#include "Frame/ROS2FrameComponent.h"
+#include "ROS2/ROS2Bus.h"
+#include "Utilities/ROS2Names.h"
+#include "Utilities/ROS2Conversions.h"
+
+#include <AzCore/Script/ScriptTimePoint.h>
+#include <AzCore/std/smart_ptr/make_shared.h>
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+
+namespace ROS2
+{
+    namespace Internal
+    {
+        const char* kImuMsgType = "sensor_msgs::msg::Imu";
+    }
+
+    void ROS2ImuSensorComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<ROS2ImuSensorComponent, ROS2SensorComponent>()->Version(1);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<ROS2ImuSensorComponent>("ROS2 Imu Sensor", "Imu sensor component")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                        ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Game"))
+                ;
+            }
+        }
+    }
+
+    ROS2ImuSensorComponent::ROS2ImuSensorComponent()
+    {
+        PublisherConfiguration pc;
+        const AZStd::string type = Internal::kImuMsgType;
+        pc.m_type = type;
+        pc.m_topic = "imu";
+        m_sensorConfiguration.m_frequency = 10;
+        m_sensorConfiguration.m_publishersConfigurations.insert(AZStd::make_pair(type, pc));
+    }
+
+    void ROS2ImuSensorComponent::Activate()
+    {
+        ROS2SensorComponent::Activate();
+        auto ros2Node = ROS2Interface::Get()->GetNode();
+        AZ_Assert(m_sensorConfiguration.m_publishersConfigurations.size() == 1, "Invalid configuration of publishers for IMU sensor");
+
+        const auto publisherConfig = m_sensorConfiguration.m_publishersConfigurations[Internal::kImuMsgType];
+        const auto fullTopic = ROS2Names::GetNamespacedName(GetNamespace(), publisherConfig.m_topic);
+        m_imuPublisher = ros2Node->create_publisher<sensor_msgs::msg::Imu>(fullTopic.data(), publisherConfig.GetQoS());
+
+        InitializeImuMessage();
+
+        m_previousPose = GetCurrentPose();
+        m_previousTime = GetCurrentTimeInSec();
```

Comment:
While it would work as implemented, I believe it is better to add default initialization to make it clear what is happening if this changes somehow - default value is deterministic and it will be more straightforward to deduce what is wrong.
Added in the latest commit.